milli/src/search/new/exact_attribute.rs

```diff
@@ -0,0 +1,270 @@
+use roaring::{MultiOps, RoaringBitmap};
+
+use super::query_graph::QueryGraph;
+use super::ranking_rules::{RankingRule, RankingRuleOutput};
+use crate::search::new::query_graph::QueryNodeData;
+use crate::search::new::query_term::ExactTerm;
+use crate::{Result, SearchContext, SearchLogger};
+
+/// A ranking rule that produces 3 disjoint buckets:
+///
+/// 1. Documents from the universe whose value is exactly the query.
```

Comment:
relevancy change: in the previous implementation, the first 2 heuristics were only run on the initial query with all the words, whereas in the new implementation, they will run on queries modified by a prior invocation of the `words` ranking rule, e.g. removing words at the end of the query.

If the invocation resulted in "hole" in the query (e.g. frequency removal), then the 2 heuristics will not discriminate documents. Otherwise, however, it will run again so that:

- Query: Batman the dark knight rises
- Word iter 1: Batman the dark knight rises
- Exactness iter 1: Batman the dark knight rises
- Exactness iter 1: Batman the dark knight rises
- Word iter 2: Batman the dark knight
- Exactness iter 2: Batman the dark knight
- Exactness iter 3: Batman the dark knight crashes